src/oracle/monitor.js

```diff
@@ -0,0 +1,429 @@
+import { Command } from 'commander';
+import { inspect } from 'util';
+import {
+    boardSlottingMarshaller,
+    makeRpcUtils,
+    storageHelper,
+    networkConfig
+} from '../lib/rpc.js';
+import {
+    makeFollower,
+    makeLeader,
+} from '@agoric/casting';
+import { validUrl, readJSONFile, saveJSONDataToFile } from '../helpers/utils.js'
+import { getCurrent } from '../lib/wallet.js';
+import { Registry, Gauge } from 'prom-client';
+import { createServer } from 'http';
+import { parse } from 'url';
+import { iterateReverse } from '@agoric/casting';
+
+const { 
+    PORT = '3001', 
+    POLL_INTERVAL = '10', 
+    AGORIC_NET, 
+    AGORIC_RPC = "http://0.0.0.0:26657" ,
+    STATE_FILE = "data/monitoring_state.json",
+    ORACLE_FILE = "config/oracles.json",
+} = process.env;
+
+assert(!isNaN(Number(PORT)), '$PORT is required');
+assert(!isNaN(Number(POLL_INTERVAL)), '$POLL_INTERVAL is required');
+assert(validUrl(AGORIC_RPC), '$AGORIC_RPC is required');
+assert(AGORIC_NET != "" && AGORIC_NET != null, '$AGORIC_NET is required');
+
+// Create a Registry which registers the metrics
+const register = new Registry()
+
+// Add a default label which is added to all metrics
+register.setDefaultLabels({
+    app: 'agoric-cl-oracle-monitor'
+})
+
+//Create gauge for value
+const oracleSubmission = new Gauge({
+    name: 'oracle_latest_value',
+    help: 'Latest value submitted by oracle',
+    labelNames: ['oracleName', 'oracle', 'feed']
+})
+
+//Create gauge for timestamp
+const oracleObservation = new Gauge({
```

Comment:
oracleLastEpoch for consistency?

Reply:
Fixed in https://github.com/jacquesvcritien/agoric-cl-middleware/commit/488fb2f82d584bd0cdbee2d0e326912c690f018e